Networking/Networking/Model/Customer.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import Codegen
+
+/// Represents a Customer entity:
+/// https://woocommerce.github.io/woocommerce-rest-api-docs/#customer-properties
+///
+public struct Customer: Codable, GeneratedCopiable, GeneratedFakeable {
+
+    /// Unique identifier for the customer
+    public let customerID: Int64
+
+    /// The email address for the customer
+    public let email: String
+
+    /// Customer first name
+    public let firstName: String?
+
+    /// Customer last name
+    public let lastName: String?
+
+    /// List of billing address data
+    public let billing: Address?
+
+    /// List of shipping address data
+    public let shipping: Address?
+
+    /// Customer struct initializer
+    ///
+    public init(customerID: Int64,
+                email: String,
+                firstName: String?,
+                lastName: String?,
+                billing: Address?,
+                shipping: Address?) {
+        self.customerID = customerID
+        self.email = email
+        self.firstName = firstName
+        self.lastName = lastName
+        self.billing = billing
+        self.shipping = shipping
+    }
+
+    /// Public initializer for the Customer
+    ///
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let customerID = try container.decode(Int64.self, forKey: .customerID)
+        let email = try container.decode(String.self, forKey: .email)
+        let firstName = try container.decodeIfPresent(String.self, forKey: .firstName) ?? ""
+        let lastName = try container.decodeIfPresent(String.self, forKey: .lastName) ?? ""
```

Comment:
Why the default strings here, given it's an optional property? I think you'd be best off with one or the other: make the property non-optional, or make it nil when we don't have the name.

Reply:
Oh that's right! `firstName` and `lastName` are not required by the API so initially I made them optional, that said, without the name we won't be able to perform a search later on. For now I'll leave them Optional for simplicity, but most likely we'll have to loop back and modify this when we implement the actual search. Removed default values at f5c1c3e69305534f9b179ffb54a0c880844f1b79